libnmstate/nm/ipv4.py

```diff
@@ -18,6 +18,7 @@
 import socket
 
 from . import nmclient
+from libnmstate import metadata
```

Comment:
As long as metadata does not  depend on `ipv4.py`. I think we are good in coding.

But for independent nm package, I can move that metadata constant to `nm/route.py` is that is what you are asking.

Reply:
As it stands now, if I understand correctly, the metadata constants are used because in your next changes, that module will add the information which is used here.
So the dependency is real... what the metadata module adds, the nm sub-package uses.
We need to break this somehow...
Something like this (it is just an example, not sure this is the solution):
```
# This is the client (like the test) calling ipv4.create_setting
ip_setting = ipv4.create_setting(ipv4_state, ipv4_metadata, ...)
```
This makes the caller responsibility to fetch the metadata.
There is also an option for something that is responsible to get a nmstate state and translate it to something that ipv4 understands...

As this is a generic issue, not limited to this PR, we can handle it independently.

Reply:
let's discuss that in another PR.